torch/_meta_registrations.py

```diff
@@ -404,8 +404,12 @@ def _linalg_svd_meta(
 
         V_shape = batch_dims + [n if full_matrices else k, n]
         V = A.new_empty(V_shape)
-        # TODO: need to distinguish cuSOLVER case? (see original code)
-        V.as_strided_(V_shape, make_contiguous_strides_for(V_shape, row_major=False))
+        # NB: This checks for CUDA since there is no way to check for cuSolver.
+        # Also, this might not work correctly on CPU when fake_device is not
+        # available as device_hint just defaults to CUDA in that case. See
+        # _linalg_svd meta in core.
+        is_cuda = device_hint(A) == "cuda"
+        V.as_strided_(V_shape, make_contiguous_strides_for(V_shape, row_major=is_cuda))
```

Comment:
Fixes issues with strides when `fake_device` is available or when tests run only on CUDA. Probably won't work in other cases (e.g., in `test_meta`) because `fake_device` would be just hardcoded to CUDA in that case (as explained in the comment). Similar to `_svd_meta` in prims.

This one also doesn't have the `conj` fix as `_svd_meta`. No existing tests failed because of this and the ones in `test_meta` wouldn't work because we cannot check the device (as mentioned above).

Reply:
cuSolver test would be better but I guess you would have to add more APIs for it (similar to how convolution has a select function binding). This is fine for now.